### PR TITLE
[Docs] Update migrate behaviour with respect to drop_table in spark-procedures docs.

### PR DESCRIPTION
**Pull request by sririshindra**

### CODE REVIEW

**Comment by sririshindra** [comment]
@samredai @szehon-ho Please take a look at this docs PR when you get a chance. This doc change corresponds to https://github.com/apache/iceberg/pull/5622 which recently got merged.

File: docs/spark-procedures.md [hunk 1]
```diff
@@ -421,12 +421,18 @@ Existing data files are added to the Iceberg table's metadata and can be read us
...
 #### Usage
 
-| Argument Name | Required? | Type | Description |
-|---------------|-----------|------|-------------|
-| `table`       | ✔️  | string | Name of the table to migrate |
-| `properties`  | ️   | map<string, string> | Properties for the new Iceberg table |
+| Argument Name | Required? | Type | Description                                                                          |
+|---------------|---------|------|--------------------------------------------------------------------------------------|
+| `table`       | ✔️ | string | Name of the table to migrate                                                         |
+| `properties`  | ️ | map<string, string> | Properties for the new Iceberg table                                                 |
```
Comment (rdblue) [inline docs/spark-procedures.md:433]: Can you revert the changes to the other lines? I don't think these were updated so there is no need to modify them.

File: docs/spark-procedures.md [hunk 2]
```diff
@@ -421,12 +421,18 @@ Existing data files are added to the Iceberg table's metadata and can be read us
...
+| `drop_backup` |   | boolean | When true, the backup table is dropped after succesful migration (defaults to false) |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:434]: I feel the 'successful migration' is a bit too redundant, how about (to make it more concise)

``` When true, the original table will not be retained as backup (defaults to false) ```

**Review by szehon-ho — COMMENTED** [review]
Some suggestions

File: docs/spark-procedures.md (hunk 1 above)
Comment (sririshindra) [reply docs/spark-procedures.md]: Done. fixed in the latest commit.

File: docs/spark-procedures.md [hunk 3]
```diff
@@ -421,12 +421,18 @@ Existing data files are added to the Iceberg table's metadata and can be read us
 
 To leave the original table intact while testing, use [`snapshot`](#snapshot) to create new temporary table that shares source data files and schema.
 
+Migrate will create a backup table with name [`table__BACKUP__`]. If you feel confident that the migration succeeded
```
[the remark this answers is not in the document]
Comment (sririshindra) [reply docs/spark-procedures.md]: Done. fixed in the latest commit.

File: docs/spark-procedures.md (hunk 2 above)
Comment (sririshindra) [reply docs/spark-procedures.md]: Done. fixed in the latest commit.

File: docs/spark-procedures.md [hunk 4]
```diff
@@ -421,12 +421,17 @@ Existing data files are added to the Iceberg table's metadata and can be read us
...
+By default, the original table is retained with the name `table_BACKUP_`. You can also explicitly pass  `drop_backup => true` 
+to drop the backup after migration finishes.
+
 #### Usage
 
 | Argument Name | Required? | Type | Description |
 |---------------|-----------|------|-------------|
 | `table`       | ✔️  | string | Name of the table to migrate |
-| `properties`  | ️   | map<string, string> | Properties for the new Iceberg table |
+| `properties`  | ️   | map<string, string> | Properties for the new Iceberg table |
```
Comment (sririshindra) [inline docs/spark-procedures.md:432]: Sorry for the multiple commits. I am unable to figure out what changed in this line. Seems exactly the same to my eyes. Something hidden character must have been added somehow.

File: docs/spark-procedures.md [hunk 5]
```diff
@@ -421,12 +421,17 @@ Existing data files are added to the Iceberg table's metadata and can be read us
 
 To leave the original table intact while testing, use [`snapshot`](#snapshot) to create new temporary table that shares source data files and schema.
 
+By default, the original table is retained with the name `table_BACKUP_`. You can also explicitly pass  `drop_backup => true`
```
Comment (szehon-ho) [inline docs/spark-procedures.md:424]: Sorry i should have been more clear, can we remove the second sentence as well?  (as its duplicated already in the flag description).  As you already mention 'by default', I think the user will know to look below.

File: docs/spark-procedures.md (hunk 4 above)
Comment (szehon-ho) [reply docs/spark-procedures.md:432]: Do you have one space after the last | ?  You can maybe just copy and paste from the previous version?

**Review by szehon-ho — COMMENTED** [review]
Looks better, just some small things

File: docs/spark-procedures.md (hunk 5 above)
Comment (sririshindra) [reply docs/spark-procedures.md]: Done, removed the second sentence.

File: docs/spark-procedures.md (hunk 4 above)
Comment (sririshindra) [reply docs/spark-procedures.md]: Yeah, I think there was an extra space after | . I removed it in the latest commit.

**Review by szehon-ho — APPROVED** [review]

**Comment by szehon-ho** [comment]
Merged, thanks @sririshindra
